pkg/operator/ceph/cluster/osd/topology/topology.go

```diff
@@ -148,3 +148,64 @@ func GetDefaultTopologyLabels() string {
 
 	return strings.Join(Labels, ",")
 }
+
+// CheckTopologyConflicts verifies that:
+// 1. No child domain (e.g. rack) has fewer distinct values than its parent (e.g. zone).
+// 2. No topology value is used under more than one label key.
+func CheckTopologyConflicts(nodes []corev1.Node) error {
```

Comment:
@arttor have you had the chance to review this discussion/plan to let us know whether you think the custom hostname label affects or is supported by our plans here?

Reply:
sorry for the delay, i was on PTO. The PR looks good and should not affect custom hostname logic.